### PR TITLE
fix: add object notification filtering

### PR DESCRIPTION
**Pull request by garrensmith**
Fixes https://linear.app/tigrisdata/issue/TIG-2941/docs-for-filtering-on-object-notifications

### CODE REVIEW

File: docs/objects/query-metadata.md [hunk 1]
```diff
@@ -31,6 +34,16 @@ Query supports following comparison operators:
 4. `>=` and `<=`: Greater than or equal and less than or equal
 5. `AND`: Combine multiple conditions in a query.
 
+## In Memory Operations
```
Comment (ovaistariq) [inline docs/objects/query-metadata.md:37]: Do we have to differentiate between SQL and In Memory operations? That might confuse the user. From their perspective it is all the same

File: docs/buckets/object-notifications.md [hunk 2]
```diff
@@ -104,6 +104,24 @@ An example notification payload is:
 }
 ```
 
+## Filtering
+
+Object notifications support adding a SQL-like filter to only receive
```
Comment (ovaistariq) [inline docs/buckets/object-notifications.md:109]: Could you rephrase the content here? There is some duplication.

File: docs/objects/query-metadata.md (hunk 1 above)
Comment (garrensmith) [reply docs/objects/query-metadata.md:37]: My aim was to make sure a user knew they needed to combine these with another filter.

File: docs/buckets/object-notifications.md (hunk 2 above)
Comment (garrensmith) [reply docs/buckets/object-notifications.md]: I've updated it. This should be clearer

**Review by ovaistariq — APPROVED** [review]
